gunicorn/tests/conftest.py

```diff
@@ -0,0 +1,93 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+import subprocess
+import pytest
+import os
+import tempfile
+import requests
+import shutil
+import logging
+import time
+
+from common import FIXTURES, PROC_NAME
+
+log = logging.getLogger('test_gunicorn')
+
+
+@pytest.fixture(scope="session")
+def setup_gunicorn(request):
+    gunicorn_tmpdir = tempfile.mkdtemp()
+    app_dir = os.path.join(gunicorn_tmpdir, 'app')
+    venv_dir = os.path.join(gunicorn_tmpdir, 'venv')
+    os.mkdir(app_dir)
+    os.mkdir(venv_dir)
+
+    venv_file_path = os.path.join(venv_dir, 'virtualenv.py')
```

Comment:
You can just require `virtualenv` in `requirements-dev.txt`  and use it like https://github.com/DataDog/integrations-core/blob/master/directory/tests/test_bench.py#L14. Then you can remove most of the logic below.